### ISSUE TITLE
Kubelet High CPU Usage

### ISSUE THREAD

**Issue by ichekrygin** [kubernetes/kubernetes#54448]
kind bug

**What happened**:
`kube-kubelet` process runs with high (300%) CPU usage, resulting in delayed handing of pod creation and termination (possibly other issues as well). 

**What you expected to happen**:
`kube-kubelet` runs with "normal" CPU usage. Granted I don't know what "normal" should be.

**How to reproduce it (as minimally and precisely as possible)**:
Start `kube-kubelet` and let it run for few days. I have observed high CPU usage (300%+) on `7d` old processes.

**Anything else we need to know?**:
I suspect this issue is somehow `containerd` related. Here is what I've tried:
- restart `kube-kubelet` - no effect, cpu climbs right back
- restart `docker` - no effect, cpu doesn't even drop
- restart `containerd`: after `kube-kubelet` comes back (restarted as a result of `containerd` restart) - cpu stays reasonably low `10%-20%`.

**Environment**:
- Kubernetes version (use `kubectl version`): `v1.7.6`
- Cloud provider or hardware configuration**: `AWS` running on 40 CPU Core instances
- OS (e.g. from /etc/os-release): CoreOS 1465.8.0 (Ladybug)
- Kernel (e.g. `uname -a`):`Linux ip-10-72-146-53.ec2.internal 4.12.14-coreos #1 SMP Wed Sep 20 22:20:05 UTC 2017 x86_64 Intel(R) Xeon(R) CPU E5-2676 v3 @ 2.40GHz GenuineIntel GNU/Linux`
- Install tools: 
- Others: Attached `pprof` before and after `containerd` restart

![kubelet-before](https://user-images.githubusercontent.com/324803/31921587-043d9b2a-b825-11e7-9717-a57634d77d64.png)
![kubelet-after](https://user-images.githubusercontent.com/324803/31921593-092796fe-b825-11e7-9034-c74bfd0d7b7b.png)

Minion Sample:
```
ip-10-72-123-456.ec2.internal   Ready     7d        v1.7.6 %CPU %MEM 5.5  0.1 <--
ip-10-72-123-567.ec2.internal   Ready     7d        v1.7.6 %CPU %MEM 330  0.1
ip-10-72-123-678.ec2.internal   Ready     7d        v1.7.6 %CPU %MEM 307  0.1
ip-10-72-123-123.ec2.internal   Ready     7d        v1.7.6 %CPU %MEM 4.4  0.1 <--
ip-10-72-123-234.ec2.internal   Ready     7d        v1.7.6 %CPU %MEM 316  0.1
ip-10-72-123-345.ec2.internal   Ready     7d        v1.7.6 %CPU %MEM 4.4  0.1 <--
```

**Comment by ichekrygin** [comment]
This could be somewhat related to https://github.com/kubernetes/kubernetes/issues/15528

**Comment by dashpole** [comment]
This looks like https://github.com/kubernetes/kubernetes/pull/49885.  You can see one the red path is getting "udp stats from proc".  I still need to cherrypick it back to 1.7, but the fix is in 1.8.

**Comment by dashpole** [comment]
This was cherrypicked back to 1.7 a while ago and should be in the latest 1.7 patch release
